assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -1978,6 +1998,27 @@ protected String build_student_view_assignment_context(VelocityPortlet portlet,
 
     } // build_student_view_assignment_context
 
+    protected String build_student_view_assignment_honorPledge_context(final VelocityPortlet portlet, final Context context, final RunData data, final SessionState state) {
+        ParameterParser params = data.getParameters();
+        String assignmentReference = (String) state.getAttribute(VIEW_SUBMISSION_ASSIGNMENT_REFERENCE);
+        String contextString = (String) state.getAttribute(STATE_CONTEXT_STRING);
+        User u = (User) state.getAttribute(STATE_USER);
+        String template = getContext(data).get("template");
+
+        Assignment a = getAssignment(assignmentReference, "build_student_view_assignment_honorPledge_context", state);
+        if (a != null && assignmentService.canSubmit(contextString, a)) {
```

Comment:
Would it be worthwhile to move this `null` check into `assignmentService.canSubmit()`?